ui/app/pages/swaps/swaps-gas-customization-modal/swaps-gas-customization-modal.container.js

```diff
@@ -0,0 +1,154 @@
+  calcGasTotal,
+  isBalanceSufficient,
+} from '../../send/send.utils'
+import SwapsGasCustomizationModalComponent from './swaps-gas-customization-modal.component'
+
+const mapStateToProps = (state) => {
+  const currentCurrency = getCurrentCurrency(state)
+  const conversionRate = getConversionRate(state)
+
+  const { modalState: { props: modalProps } = {} } = state.appState.modal || {}
+  const {
+    txParams = {},
+    customGasLimitMessage = '',
+    customTotalSupplement = '',
+    extraInfoRow = null,
+    initialGasPrice,
+    initialGasLimit,
+  } = modalProps || {}
+  const buttonDataLoading = swapGasPriceEstimateIsLoading(state)
+
+  const swapsCustomizationModalPrice = getSwapsCustomizationModalPrice(state)
+  const swapsCustomizationModalLimit = getSwapsCustomizationModalLimit(state)
+
+  const customGasPrice = swapsCustomizationModalPrice || initialGasPrice || getSwapsFallbackGasPrice(state)
+  const customGasLimit = swapsCustomizationModalLimit || initialGasLimit
+
+  const { value } = txParams
+  const customGasTotal = calcGasTotal(customGasLimit, customGasPrice)
+
+  const swapsGasPriceEstimates = getSwapGasPriceEstimateData(state)
+
+  const {
+    averageEstimateData,
+    fastEstimateData,
+  } = getRenderableGasButtonData(swapsGasPriceEstimates, customGasLimit, true, conversionRate, currentCurrency)
+  const gasButtonInfo = [averageEstimateData, fastEstimateData]
+
+  const newTotalFiat = sumHexWEIsToRenderableFiat(
+    [value, customGasTotal, customTotalSupplement],
+    currentCurrency,
+    conversionRate,
+  )
+
+  const balance = getCurrentEthBalance(state)
+
+  const newTotalEth = sumHexWEIsToRenderableEth([value, customGasTotal, customTotalSupplement])
+
+  const sendAmount = sumHexWEIsToRenderableEth([value, '0x0'])
+
+  const insufficientBalance = !isBalanceSufficient({
+    amount: value,
+    gasTotal: customGasTotal,
+    balance,
+    conversionRate,
+  })
+
+  return {
+    customGasPrice,
+    customGasLimit,
+    averageGasPriceInHexWei: averageEstimateData?.priceInHexWei,
+    customPriceIsSafe: !isCustomSwapsGasPriceUnSafe(state),
+    gasPriceButtonGroupProps: {
+      buttonDataLoading,
+      defaultActiveButtonIndex: getDefaultActiveButtonIndex(gasButtonInfo, customGasPrice),
+      gasButtonInfo,
+    },
+    infoRowProps: {
+      originalTotalFiat: sumHexWEIsToRenderableFiat(
+        [value, customGasTotal, customTotalSupplement],
+        currentCurrency,
+        conversionRate,
+      ),
+      originalTotalEth: sumHexWEIsToRenderableEth(
+        [value, customGasTotal, customTotalSupplement],
+      ),
+      newTotalFiat,
+      newTotalEth,
+      transactionFee: sumHexWEIsToRenderableEth(['0x0', customGasTotal]),
+      sendAmount,
+      extraInfoRow,
+    },
+    gasEstimateLoadingHasFailed: swapGasEstimateLoadingHasFailed(state),
+    insufficientBalance,
+    customGasLimitMessage,
+    customTotalSupplement,
+    conversionRate,
+    value,
+    disableSave: insufficientBalance,
```

Comment:
We should probably also disable save in the event that gas price, gas limit, or both are unset. Or is this already handled somehow?

Reply:
The gas input buttons default to 0, so this is not possible

Reply:
Huh, I see. We can check that the custom gas limit is safe though at least, using `customPriceIsSafe`

Reply:
So desired UX is to still allow users to submit with gas prices below the average estimate, and just show a warning. In the gas-customization modal, this is all `customPriceIsSafe` is safe does. Nowhere do we disable save based on gas price.

I have renamed this variable and associated function in `swaps-gas-customization-modal` to make its purpose clearer in 65db04b8e

Reply:
Meanwhile, https://github.com/MetaMask/metamask-extension/pull/9600 ensures that save is disabled if gas limit is too low.

Reply:
Ah, sorry I was getting `customPriceIsSafe` mixed up with the minimum custom limit from that PR.